### PR TITLE
64-bit linux support for v0.43.05

### PR DESCRIPTION
**Pull request by DanFritz**
Tested against v0.43.05 on Archlinux x86_64 system

### CODE REVIEW

File: inc/utils.h [hunk 1]
```diff
@@ -31,10 +31,9 @@ THE SOFTWARE.
 #include <QString>
 #include <math.h>
 
-// valid for as long as DF stays 32bit
-typedef quint32 VIRTADDR;
-typedef quint32 USIZE;
-typedef qint32 SSIZE;
+typedef quintptr VIRTADDR;
+typedef quintptr USIZE;
+typedef qintptr SSIZE;
```
Comment (carterscottm) [inline inc/utils.h]:
```
#if QT_MAJOR_VERSION < 5
    typedef qint64 qintptr;
#endif
typedef qintptr SSIZE;
```
Will allow this project to be built with Qt4 and therefore pass travis CI

Comment (DanFritz) [reply inc/utils.h]: Thanks!

File: inc/utils.h [hunk 2]
```diff
@@ -31,6 +31,10 @@ THE SOFTWARE.
 #include <QString>
 #include <math.h>
 
+#if QT_MAJOR_VERSION < 5
+  typedef qint64 quintptr
```
Comment (lethosor) [inline inc/utils.h]: Missing a semicolon here

File: src/squad.cpp [hunk 3]
```diff
@@ -129,7 +129,7 @@ void Squad::read_members() {
 
         m_uniforms.insert(position,u);
         LOGD << "checking orders for position" << position;
-        foreach(VIRTADDR ord_addr, m_df->enumerate_vector(addr+0x4)){ //TODO: offset
+        foreach(VIRTADDR ord_addr, m_df->enumerate_vector(addr+0x2)){ //TODO: offset
```
Comment (lethosor) [inline src/squad.cpp:132]: Is this something that should be correct in the layouts instead?

File: inc/utils.h (hunk 2 above)
Comment (carterscottm) [reply inc/utils.h]: Also, it should be qintptr instead of quintptr

**Comment by bendlas** [comment]
Tried to build this and got a build failure: https://gist.github.com/bendlas/7c5cfbca2c8970a9457c4f20394bf6b3

**Comment by Abkaal** [comment]
Crashes for me whenever I try to read a fort that has a squad. Log:
2017-Aug-12 15:40:51.805 INFO	core	FOUND 134 squads [src/dfinstance.cpp:873] (load_squads)
Qt has caught an exception thrown from an event handler. Throwing
exceptions from an event handler is not supported in Qt. You must
reimplement QApplication::notify() and catch all exceptions there.

terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
Aborted

I'm sure I had 2 squads at this point. Crashed when I had only 1 as well.

**Comment by somoking** [comment]
Can confirm Abkaals crash. If a unit is assigned to a squad then DF Therapist will crash squads themselves will cause no issue.

File: src/squad.cpp (hunk 3 above)
Comment (Nemoder) [reply src/squad.cpp:132]: Changing this back to addr+0x4 fixes the squad crash on startup for me.

**Pull request closed without merging**